chainerx_cc/chainerx/routines/arithmetic.cc

```diff
@@ -491,4 +491,34 @@ Array Power(const Array& x1, Scalar x2) { return internal::Binary(&PowerASImpl,
 
 Array Power(Scalar x1, const Array& x2) { return internal::Binary(&PowerSAImpl, x1, x2, GetArithmeticResultDtype(x1, x2)); }
 
+Array Fmod(const Array& x1, const Array& x2) {
+    CheckEqual(x1.shape(), x2.shape());
+    Array out = Empty(x1.shape(), x1.dtype(), x1.device());
+
+    {
+        NoBackpropModeScope scope{};
+        x1.device().backend().CallKernel<FmodKernel>(x1, x2, out);
+    }
+
+    {
+        BackwardBuilder bb{"fmod", x1, x2};
+        if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+            bt.Define([dtype = x1.dtype()](BackwardContext& bctx) {
+                const Array& gx = *bctx.output_grad();
```

Comment:
```suggestion
                Array gx = *bctx.output_grad();
```

Reply:
I guess
`const Array& gx = *bctx.output_grad();`
Given that we don't move below.